internal/common/conversion/type_conversion.go

```diff
@@ -56,3 +56,17 @@ func IsStringPresent(strPtr *string) bool {
 func MongoDBRegionToAWSRegion(region string) string {
 	return strings.ReplaceAll(strings.ToLower(region), "_", "-")
 }
+
+func SlicePtrToSlice[T any](v *[]T) []T {
+	if v == nil {
+		return nil
+	}
+	return *v
+}
+
+func NonEmptySliceToPtrSlice[T any](v []T) *[]T {
```

Comment:
@AgustinBettati idea to keep same current behavior

Reply:
curious: why are you specifying `NonEmpty...` in the method name? the name makes me thinking you can't call the method for an empty slice, which in reality you can

Reply:
it's to highlight that it will return nil for empty slice instead of a pointer to a empty slice.  This is to keep current behaviour.    (normally you would expect to return the pointer to the slice even if empty but not nil)

we want to send nil pointer for empty slices so it's not send in JSON to the API as current behavior.

Reply:
ok I get it.